### PR TITLE
fix(es/minifier): Collect raw str values for new Tpl element

### PR DESCRIPTION
**Pull request by stormslowly**
**Description:**
convert `raw` value  to `cooked` for creating new Tpl element.

<del>
while optimizing nested tpl, after concatenating some cooked strs to assemble a new tpl element,
the `raw` of the tpl should be escaped cooked string value.
</del>

**BREAKING CHANGE:**

None

**Related issue (if exists):**
https://github.com/swc-project/swc/issues/9184

### CODE REVIEW

**Comment by stormslowly** [comment]
@kdy1  Oops, a compress case failed, i am trying to fix

File: crates/swc_ecma_minifier/tests/exec.rs [hunk 1]
```diff
@@ -11325,3 +11325,13 @@ fn issue_9010() {
         "#,
     );
 }
+
+#[test]
+fn issue_9184() {
+    run_default_exec_test(
+        r#"
+        let pi= Math.random() >1.1 ? "foo": "bar";
```
Comment (kdy1) [inline crates/swc_ecma_minifier/tests/exec.rs:11333]: Can we have one more case for the reverse case? (`Math.random() < -1`)

Comment (stormslowly) [reply crates/swc_ecma_minifier/tests/exec.rs:11333]: I am not quite understand why adding the reverse case.
IMO, the key of this case is about  purify the nested tpl string. 
The `Math.random()` is just for the `pi` variable not being to replaced by its literal value in the tpl string.

Comment (kdy1) [reply crates/swc_ecma_minifier/tests/exec.rs:11333]: Ah, ok

File: crates/swc_ecma_minifier/src/compress/pure/strings.rs [hunk 2]
```diff
@@ -121,13 +121,15 @@ impl Pure<'_> {
...
-        let mut cur_str_value = String::new();
+        let mut cur_cooked_str = String::new();
+        let mut cur_raw_str = String::new();
 
         for idx in 0..(tpl.quasis.len() + tpl.exprs.len()) {
             if idx % 2 == 0 {
                 let q = tpl.quasis[idx / 2].take();
 
-                cur_str_value.push_str(q.cooked.as_deref().unwrap_or(&*q.raw));
+                cur_cooked_str.push_str(q.cooked.as_deref().unwrap_or(&*q.raw));
```
Comment (kdy1) [inline crates/swc_ecma_minifier/src/compress/pure/strings.rs:131]: I think this logic is wrong. `cooked` or `raw` should never be the case, and we should run tpl string replace logic for `raw` if we want to use it as a `value`.

See https://rustdoc.swc.rs/swc_ecma_ast/struct.Str.html#method.from_tpl_raw

Comment (stormslowly) [reply crates/swc_ecma_minifier/src/compress/pure/strings.rs]: updated

**Review by kdy1 — APPROVED** [review]
Thanks!

---

swc-bump:
 - swc_ecma_minifier --breaking

**Pull request merged**